packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -428,12 +388,25 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
         }
     }
 
-    private getTimeX(event: number): number {
-        const offset = this.props.viewRange.getOffset() ?? 0;
-        const scale = this.props.viewRange.getEnd() - this.props.viewRange.getstart();
-        const xPos = this.props.viewRange.getstart() - offset +
-            (event / this.lineChartRef.current.chartInstance.width) * scale;
-        return xPos;
+    private getTimeForX(x: number): bigint {
+        if (!this.lineChartRef.current?.chartInstance) {
+            return BigInt(0);
+        }
+        const offset = this.props.viewRange.getOffset() ?? BigInt(0);
+        const duration = this.props.viewRange.getDuration();
+        const time = this.props.viewRange.getStart() - offset +
+            BIMath.round(x / this.lineChartRef.current.chartInstance.width * Number(duration));
```

Comment:
It's required because you can't mix Number and BigInt in calculations. I guess we could change the other two values to BigInt and change to the order to do integer math, but then it would be a floored result.

I think both options are OK. The loss of precision occurs when the view range is very large, and at that zoom level, a mouse x position is inherently a very coarse time selection.

Just a question, if we use BigInt integer math instead, we have to be very careful to always multiply first and divide last. Is this a potential for error that is best avoided?

Reply:
I think it's ok.